File: dotnetcore/PritsetClient.cs
using System.Text.Json.Nodes;

namespace PritsetDotnetExample;

public static class PritsetClient
{
    public static PritsetRequest CreateRequest(string templateId)
    {
        var json = File.ReadAllText(Path.Combine("Data", "dummy_data.json"));
        var data = JsonNode.Parse(json)
            ?? throw new InvalidOperationException("Unable to parse Data/dummy_data.json.");

        return new PritsetRequest
        {
            Data = data,
            Token = "your-access-token-here-from-https://app.pritset.com/settings",
            Secret = "your-secret-here-from-https://app.pritset.com/settings",
            Api = $"https://api.pritset.com/api/template/process/direct/{templateId}",
        };
    }
}

File: java/PritsetClient.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public final class PritsetClient {
    private PritsetClient() {
    }

    public static PritsetRequest createRequest(String templateId) throws IOException {
        String data = Files.readString(Path.of("Data", "dummy_data.json"));

        return new PritsetRequest(
            data,
            "your-access-token-here-from-https://app.pritset.com/settings",
            "your-secret-here-from-https://app.pritset.com/settings",
            "https://api.pritset.com/api/template/process/direct/" + templateId
        );
    }
}

